kinbot/reaction_generator.py

```diff
@@ -195,51 +195,46 @@ def generate(self):
                      
                     for i, st_pt in enumerate(obj.products_final):
                         chemid = st_pt.chemid
-                        orig_geom = copy.deepcopy(st_pt.geom)
                         e, st_pt.geom = self.qc.get_qc_geom(str(st_pt.chemid) + '_well', st_pt.natom)
                         if e < 0:
-                            logging.info('\tProduct optimization failed for {}, product {}'.format(instance_name,st_pt.chemid))
+                            logging.info('\tProduct optimization failed for {}, product {}'.format(instance_name, st_pt.chemid))
                             self.species.reac_ts_done[index] = -999
                             err = -1
                         elif e != 0:
                             err = -1
                         else:
                             e2, st_pt.energy = self.qc.get_qc_energy(str(st_pt.chemid) + '_well')
                             e2, st_pt.zpe = self.qc.get_qc_zpe(str(st_pt.chemid) + '_well')
-                            st_pt.bond_mx()
                             st_pt.characterize(0)  # not allowed to use the dimer option here
-                            st_pt.calc_chemid()
                             if chemid != st_pt.chemid:
                                 obj.products_final.pop(i)
-                                newfrags, newmaps = st_pt.start_multi_molecular()
+                                newfrags, newmaps = st_pt.start_multi_molecular()  # newfrags is list of stpt obj 
                                 products_waiting_status[index] = [0 for frag in newfrags]
                                 fragChemid=[]
-                                for i, a in enumerate(newfrags):
+                                for i, newfr in enumerate(newfrags):
                                     for prod in frag_unique:
-                                        if a.chemid == prod.chemid:
+                                        if newfr.chemid == prod.chemid:
                                             newfrags.pop(i)
-                                            a=prod
-                                            j=i-1
-                                            newfrags.insert(j,a)
+                                            newfr = prod
+                                            j = i - 1
+                                            newfrags.insert(j, newfr)
                                     #add new frag to frag_unique somehow?
-                                    j=i-1
-                                    obj.products_final.insert(j,a)
-                                    self.qc.qc_opt(a, a.geom, 0)
-                                    fragChemid.append(a.chemid)
+                                    j = i - 1
+                                    obj.products_final.insert(j, newfr)
+                                    self.qc.qc_opt(newfr, newfr.geom, 0) # what does this do?
```

Comment:
This should be optimizing the new fragment, because as far as I could tell within the code the new fragments were created based on the geometry, but they still needed to be optimized.

Reply:
I'm not sure what you are reacting to here, let's talk off-line

Reply:
OK